quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -38,12 +43,21 @@ import { lastTraceName } from '../compile'
  * the type checker yet, computations may fail with weird JavaScript errors.
  */
 export class CompilerVisitor implements IRVisitor {
+  // the id of the current module
+  private moduleId: bigint = 0n
+  // the lookup table to use for the module
+  private lookupTable: LookupTable = newTable({})
+  // the scope tree to be used with the lookup table
+  private scopeTree: ScopeTree = { value: 0n, children: [] }
+  // types assigned to expressions and definitions
+  private types: Map<bigint, TypeScheme>
```

Comment:
Are we using this value for something?

Reply:
Interestingly, not :)

We would need it for higher-order definitions though. So I would keep it